lang/typescript/README.md

```diff
@@ -1,6 +1,68 @@
 # Worldwide - Typescript
 
-🚧 Still in development -- not ready for use yet 🚧
+Utilities for parsing and formatting address fields
+
+## Usage
+
+### Generating a concatenated address string
+
+To generate a concatenated string for a specific address field, pass an address
+object with the country code and extended fields as defined in the region yaml.
+
+For example, the Brazil yaml defines the format as follows:
+
+```yaml
+# db/data/region/BR.yml
+combined_address_format:
+  address1:
+    - key: streetName
+    - key: streetNumber
+      decorator: ","
+  address2:
+    - key: line2
+    - key: neighborhood
+      decorator: ","
+```
+
+To generate a correctly formatted `address1` string for Brazil, include
+`streetName` and `streetNumber` in the address object.
+
+```ts
+generateAddress1({
+  countryCode: 'BR',
+  streetName: 'Av. Paulista',
+  streetNumber: '1578',
+}); // returns 'Av. Paulista,\u00A0 1578'
+```
+
+You can generate Address1 or Address2 fields for any supported country, even if
```

Comment:
I believe generateAddress1 returns null if the country does not define a concatenated format. Maybe we could mention that we return an unmodified address1 value if it is defined in the source address?